test/neural_networks/test_sampler_qnn.py

```diff
@@ -0,0 +1,182 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Sampler QNN with Terra primitives."""
+import numpy as np
+from test import QiskitMachineLearningTestCase
+
+from qiskit.primitives import Sampler
+from qiskit.algorithms.gradients import ParamShiftSamplerGradient, FiniteDiffSamplerGradient
+from qiskit.circuit.library import RealAmplitudes, ZZFeatureMap
+from qiskit.utils import QuantumInstance, algorithm_globals
+from qiskit import Aer
+from qiskit.utils import QuantumInstance
+
+from qiskit_machine_learning.neural_networks import CircuitQNN
+from qiskit_machine_learning.neural_networks.sampler_qnn import SamplerQNN
+
+algorithm_globals.random_seed = 42
+from test.connectors.test_torch import TestTorch
+import os
+
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
+
+class TestSamplerQNN(QiskitMachineLearningTestCase):
+    """Sampler QNN Tests."""
+
+    def setUp(self):
+        super().setUp()
+        algorithm_globals.random_seed = 12345
+
+        # define test circuit
+        num_qubits = 3
+        self.qc = RealAmplitudes(num_qubits, entanglement="linear", reps=1)
+        self.qi_qasm = QuantumInstance(Aer.get_backend("aer_simulator"))
+        self.sampler = Sampler()
+
+    def test_forward_pass(self):
+
+        parity = lambda x: "{:b}".format(x).count("1") % 2
+        output_shape = 2  # this is required in case of a callable with dense output
+
+        circuit_qnn = CircuitQNN(
```

Comment:
Will this be reduced to just the Sampler QNN testing - it looks like you were comparing new/old here but in the end we already have the separate test for the CircuitQNN so this will become independent of that right - it needs to be as CircuitQNN will be removed in the future.

Reply:
I kept the test for the time being but added that it should be removed once `CircuitQNN` is fully deprecated